packages/frontend/editor-ui/src/plugins/codemirror/typescript/client/tests/completions.test.ts

```diff
@@ -0,0 +1,39 @@
+import { matchText } from '../completions';
+import { CompletionContext } from '@codemirror/autocomplete';
+import { EditorState } from '@codemirror/state';
+import { n8nLang } from '@/plugins/codemirror/n8nLang';
+
+describe('matchText', () => {
+	afterEach(() => {
+		vi.resetAllMocks();
+	});
+
+	it('should match on previous nodes', () => {
+		const previousNodes = [
+			{ doc: '$(|)', expected: '(' },
+			{ doc: '$("|', expected: '"' },
+			{ doc: "$('|", expected: "'" },
+			{ doc: '$(""|', expected: '"' },
+			{ doc: "$('|", expected: "'" },
+			{ doc: '$("|")', expected: '"' },
+			{ doc: "$('|')", expected: "'" },
+			{ doc: '$("|)', expected: '"' },
+			{ doc: '$("No|")', expected: 'No' },
+			{ doc: "$('No|')", expected: 'No' },
+			{ doc: '$("N|")', expected: 'N' },
+			{ doc: "$('N|')", expected: 'N' },
+		];
```

Comment:
You could use `it.each` instead for some cleaner test output, examples in `FilterParameter.test.ts`